napari/_qt/widgets/qt_extension2reader.py

```diff
@@ -170,16 +170,17 @@ def _filter_compatible_readers(self, new_pattern):
 
         readers = self._npe2_readers.copy()
         to_delete = []
-        compatible_readers = get_potential_readers(new_pattern)
+        if new_pattern:
+            compatible_readers = get_potential_readers(new_pattern)
+        else:
+            compatible_readers = {}
         for plugin_name in readers:
             if plugin_name not in compatible_readers:
                 to_delete.append(plugin_name)
 
         for reader in to_delete:
             del readers[reader]
 
-        readers.update(self._npe1_readers)
```

Comment:
npe1 readers are ignored by `self._add_reader_choice`, but impact if condition.

Reply:
not sure what's given you that impression? On `main` with an npe1 reader installed I see the reader both in the initial list, and after filtering with an extension. In this branch, I no longer see the npe1 reader after filtering - so I think we do need this line.

Reply:
Dummy npe1 reader for testing purposes: https://github.com/DragaDoncila/npe1-read